src/components/NcRichText/NcReferenceList.vue

```diff
@@ -118,15 +119,26 @@ export default {
 		},
 		resolve() {
 			const match = (new RegExp(URL_PATTERN).exec(this.text.trim()))
+			const isPublic = !getCurrentUser()?.uid
+			const sharingToken = document.getElementById('sharingToken')?.value || ''
 			if (this.limit === 1 && match) {
-				return axios.get(generateOcsUrl('references/resolve', 2) + `?reference=${encodeURIComponent(match[0])}`)
+				return isPublic
+					? axios.get(generateOcsUrl('references/resolvePublic', 2) + `?reference=${encodeURIComponent(match[0])}&sharingToken=${sharingToken}`)
```

Comment:
```suggestion
					? axios.get(generateOcsUrl('references/resolvePublic') + `?reference=${encodeURIComponent(match[0])}&sharingToken=${sharingToken}`)
```

https://nextcloud-libraries.github.io/nextcloud-router/functions/generateOcsUrl.html
The second param of `generateOcsUrl` is the placeholder replacement object.

Reply:
Good catch, fixed now. Apparently this was wrong already before my changes.